client/src/layers/EditAnnotationLayer.ts

```diff
@@ -350,14 +399,21 @@ export default class EditAnnotationLayer extends BaseLayer<GeoJSON.Feature> {
     if (this.featureLayer === e.annotation.layer()) {
       if (e.annotation.state() === 'done' && this.formattedData.length === 0) {
         // geoJS insists on calling done multiple times, this will prevent that
-        this.formattedData = [e.annotation.geojson()];
+        const geoJSONData = [e.annotation.geojson()];
+        if (this.type === 'rectangle') {
+          geoJSONData[0].geometry.coordinates[0] = reOrdergeoJSON(
+            geoJSONData[0].geometry.coordinates[0] as GeoJSON.Position[],
+          );
+        }
+        this.formattedData = geoJSONData;
         // The new annotation is in a state without styling, so apply local stypes
         this.applyStylesToAnnotations();
         //This makes sure the click for the end point doesn't kick us out of the mode
         this.disableModeSync = true;
+
         this.bus.$emit(
           'update:geojson',
-          'editing',
+          this.getMode(),
```

Comment:
```
        this.bus.$emit(
          'update:geojson',
          'editing',
          this.getMode() === 'creation', // signifies if the shape was just completed
          this.formattedData[0],
          this.type,
          this.selectedKey,
          this.skipNextFunc(),
        );
```

Proposal: don't add a new mode, just a flag that signifies that the shape was completed. Use that flag in `LayerManager`, but don't expose it to the rest of the app.  Revert all the changes where `creation` was added.